src/deciders/content_length.rs

```diff
@@ -0,0 +1,71 @@
+use super::{Decider, DeciderHooks};
+use crate::actions::Action;
+use crate::observers::{Observers, ResponseObserver};
+use crate::responses::Response;
+use crate::state::SharedState;
+
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
+use cfg_if::cfg_if;
+
+cfg_if! {
+    if #[cfg(docsrs)] {
+        // just bringing in types for easier intra-doc linking during doc build
+        use crate::fuzzers::Fuzzer;
+        use crate::build_deciders;
+        use crate::deciders::Deciders;
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Debug)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[allow(clippy::derive_partial_eq_without_eq)] // known false-positive introduced in 1.63.0
+pub struct ContentLengthDecider<F>
+    where
+        F: Fn(usize, usize, &SharedState) -> Action,
```

Comment:
`F: Fn(usize, usize, &SharedState) -> Action + Sync + Send + Clone + 'static`

Reply:
I don't think adding the extra constraints will break anything, but they may. If they do break other things, don't worry about it,  I'll get them in my branch.